zod/cli/torrent_download.py

```diff
@@ -0,0 +1,347 @@
+
+
+@app.command()
+def download(
+    output_dir: str = typer.Option(
+        ...,
+        help="Output directory where dataset will be extracted",
+        prompt="Where do you want to extract the dataset (e.g. ~/data/zod)?",
+        prompt_required=False,
+        rich_help_panel=REQ,
+    ),
+    subset: SubDataset = typer.Option(
+        ...,
+        help="The sub-dataset to download",
+        prompt="Which sub-dataset do you want to download?",
+        prompt_required=False,
+        rich_help_panel=REQ,
+    ),
+    version: Version = typer.Option(
+        ...,
+        help="The version of the dataset to download",
+        prompt="Which version do you want to download?",
+        prompt_required=False,
+        rich_help_panel=REQ,
+    ),
+    # General download settings
+    rm: bool = typer.Option(False, help="Remove the downloaded archives", rich_help_panel=GEN),
+    dry_run: bool = typer.Option(False, help="Print what would be downloaded", rich_help_panel=GEN),
+    extract: bool = typer.Option(True, help="Unpack the archives", rich_help_panel=GEN),
+    extract_already_downloaded: bool = typer.Option(
+        False, help="Extract already downloaded archives", rich_help_panel=GEN
+    ),
+    no_confirm: bool = typer.Option(
+        False,
+        "-y",
+        "--no-confirm/--confirm",
+        help="Don't ask for confirmation",
+        is_flag=True,
+        flag_value=False,
+        rich_help_panel=GEN,
+    ),
+    # Filter settings
+    annotations: bool = typer.Option(True, help="Download annotations", rich_help_panel=FIL),
+    images: bool = typer.Option(True, help="Whether to download the images", rich_help_panel=FIL),
+    blur: bool = typer.Option(True, help="Download blur images", rich_help_panel=FIL),
+    lidar: bool = typer.Option(True, help="Download lidar data", rich_help_panel=FIL),
+    oxts: bool = typer.Option(True, help="Download oxts data", rich_help_panel=FIL),
+    infos: bool = typer.Option(True, help="Download infos", rich_help_panel=FIL),
+    vehicle_data: bool = typer.Option(True, help="Download the vehicle data", rich_help_panel=SEQ),
+    dnat: bool = typer.Option(False, help="Download DNAT images", rich_help_panel=FRA),
+    num_scans_before: int = typer.Option(
+        0,
+        help="Number of earlier lidar scans to download (-1 == all)",
+        rich_help_panel=FRA,
+    ),
+    num_scans_after: int = typer.Option(
+        0, help="Number of later lidar scans to download (-1 == all)", rich_help_panel=FRA
+    ),
+    parallel: bool = typer.Option(
+        False,
+        help="Not used in this script. Exist for compatibility with zod/cli/download.py",
+        rich_help_panel=GEN,
+    ),
+    max_workers: int = typer.Option(
+        1,
+        help="Not used in this script. Exist for compatibility with zod/cli/download.py",
+        rich_help_panel=GEN,
+    ),
+):
+    # we have to make sure aria2c is installed
+    check_aria_install()
```

Comment:
would be nice if this check considered the dry_run flag. So that I could see what command would run even without having aria installed :)

Reply:
Good suggestion. Added!